core/src/main/java/com/inrupt/client/core/DefaultClient.java

```diff
@@ -84,6 +85,7 @@ public <T> CompletionStage<Response<T>> send(final Request request,
             .map(token -> httpClient.send(upgradeRequest(request, token), responseBodyHandler))
             // Otherwise perform the regular HTTP authorization dance
             .orElseGet(() -> httpClient.send(request, responseBodyHandler)
+                .handle((res, err) -> recover(res, err))
```

Comment:
Could `recover` be a method reference or are they not available in 8?

Reply:
I also think an inline comment here would help, just like the explanatory notes above.